### PR TITLE
[loki-stack] Bump chart versions of all dependencies of Loki-stack chart

### PR DESCRIPTION
**Pull request by martinbeentjes**
This PR bumps the Promtail version in the `loki-stack` chart to 2.4.2.

### CODE REVIEW

**Comment by zanhsieh** [comment]
@martinbeentjes 
Please sign DCO. Thank you.

**Comment by martinbeentjes** [comment]
@zanhsieh Signed DCO.

**Comment by feber** [comment]
Apologies for jumping in, I have a question.
Should the `version` here point the application version, or chart version?
Because there are PRs that updates according to chart version. [[1]](https://github.com/grafana/helm-charts/pull/1054/files#diff-157a2ab75eb36d7f141f3e314b293bef4c0a8600b211287c42376320bdd40110R9) [[2]](https://github.com/grafana/helm-charts/pull/1045/files#diff-157a2ab75eb36d7f141f3e314b293bef4c0a8600b211287c42376320bdd40110R9)

**Comment by zanhsieh** [comment]
@feber 
version = chart version
appVersion = the application itself version
@martinbeentjes 
I think you miss bumping chart version (version in Chart.yaml) to latest. Thank you.

**Comment by martinbeentjes** [comment]
@zanhsieh fixed.

**Review by zanhsieh — APPROVED** [review]

File: charts/loki-stack/requirements.yaml [hunk 1]
```diff
@@ -6,7 +6,7 @@ dependencies:
 - name: "promtail"
   condition: promtail.enabled
   repository: "https://grafana.github.io/helm-charts"
-  version: "^2.2.0"
+  version: "^2.4.2"
```
Comment (feber) [inline charts/loki-stack/requirements.yaml:9]: Perhaps the [error in pipeline](https://github.com/grafana/helm-charts/runs/5473495275?check_suite_focus=true#step:7:31) triggered by this?
Looking at the releases, there seems to be [no Chart with version 2.4.2](https://github.com/grafana/helm-charts/releases?expanded=true&page=3&q=promtail).

Comment (feber) [inline charts/loki-stack/requirements.yaml:9]: Perhaps the [error in pipeline](https://github.com/grafana/helm-charts/runs/5473495275?check_suite_focus=true#step:7:31) triggered by this?
Looking at the releases, there seems to be [no Chart with version 2.4.2](https://github.com/grafana/helm-charts/releases?expanded=true&page=3&q=promtail).

**Review by zanhsieh — APPROVED** [review]

**Comment by martinbeentjes** [comment]
@zanhsieh I saw that the version was wrong for Logstash and filebeat, should now be a valid chart version.

**Review by zanhsieh — APPROVED** [review]

**Review by zanhsieh — DISMISSED** [review]

**Comment by lc-guy** [comment]
Would be useful to have this merged -- current promtail dependency chart's default config breaks the log format (for parsing with loki datasource in grafana).

**Comment by zanhsieh** [comment]
@martinbeentjes 
Please look into the lint-test log and fix the bug. Thank you.

**Comment by alteredtech** [comment]
I figured out where the bug is but I am not sure how to implement a fix. Between promtail chart version 2.2.0 and 3.11.0, the config section was changed/added. In this section, config.lokiAddress was added. This address is put into config.file further down. In chart version 3.11.0, promtail is looking for loki-gateway as a host when it should be looking for I think loki.fullname:3100.
Basically it needs the name of the loki pod with port 3100. Unless there is a better way of doing it, when I changed my values file with lokiAddress to the name of the loki pod and port, it worked.

**Comment by feber** [comment]
Thanks for digging into this @alteredtech.
Based on your finding, my local test passed by passing the release name:
```yml
promtail:
  enabled: true
  config:
    lokiAddress: http://{{ .Release.Name }}:3100/loki/api/v1/push
```

But I am not sure whether this is the correct/recommended/acceptable way to do it or not.

**Comment by martinbeentjes** [comment]
Thanks all for helping out. I've been a bit too busy with my work to get some time to attempt a fix here. I've applied the snippet provided by @feber.

**Review by zanhsieh — APPROVED** [review]

**Comment by martinbeentjes** [comment]
@zanhsieh is there anything I need to do for those two workflows to run? I expected your approval would be enough. Hopefully I can help out getting those to run. Thanks in advance!

**Comment by zanhsieh** [comment]
@martinbeentjes 
It seems one more reviewer needs to approve. The "Merge pull request" button is gray-out in my side (can't click).

**Comment by stephaneetje** [comment]
On the page of grafana's fluent-bit helm chart it sais it is depecrated. so we can't use fluentbit with loki-stack chart anymore ??

**Comment by martinbeentjes** [comment]
@stephaneetje Good one, might be that we also need to change `charts/loki-stack/requirements.yaml` then.

```diff
 - name: "fluent-bit"
    condition: fluent-bit.enabled
-   repository: "https://grafana.github.io/helm-charts"
-   version: "^2.3.0"
+   repository: "https://fluent.github.io/helm-charts"
+   version: "^0.19.22"
```

**Comment by zdevi** [comment]
The values.yaml must also be changed. There are diferences beetwen grafanas fluentbit and the official one

**Comment by martinbeentjes** [comment]
Might be useful to create a separate PR to change this. First bump every version we are using right now. Then we remove the deprecated chart that is used.

**Comment by martinbeentjes** [comment]
I have created #1140 for this change.

**Comment by martinbeentjes** [comment]
@feber Could you do a review again, that would help out I think with getting the required approvals. Thanks in advance!

**Review by feber — APPROVED** [review]

File: charts/loki-stack/Chart.yaml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 apiVersion: "v1"
 name: loki-stack
 version: 2.6.2
```
Comment (zanhsieh) [inline charts/loki-stack/Chart.yaml:3]: need to bump to 2.6.3.

**Review by zanhsieh — CHANGES_REQUESTED** [review]

File: charts/loki-stack/Chart.yaml [hunk 3]
```diff
@@ -1,7 +1,7 @@
 apiVersion: "v1"
 name: loki-stack
-version: 2.6.2
-appVersion: v2.1.0
+version: 2.6.3
```
Comment (gcaracuel) [inline charts/loki-stack/Chart.yaml:3]:
```suggestion
version: 2.6.4
```

Needs a version bump

Comment (martinbeentjes) [reply charts/loki-stack/Chart.yaml]: @gcaracuel done

**Review by zanhsieh — APPROVED** [review]

**Comment by danports** [comment]
This PR introduces a breaking change (included in the `promtail` subchart upgrade) but only increments the `loki-stack` chart's patch version. Can we consider pulling version 2.6.4 and replacing it with a 3.0.0? See discussion in #1256.

**Pull request merged**